### PR TITLE
MNT: Use != to pin problematic Numpy 1.15.3

### LINKED ISSUE #8051: TST: Pinning max numpy version broke numpy-dev build
Relevant log: https://travis-ci.org/astropy/astropy/jobs/448451307 (the build right before the merge of #8041 was business as usual; I checked)

So, pinning it in `setup.py` breaks our numpy-dev CI, but unpinning it breaks our RTD build. @mhvk , when is Numpy 1.15.4 coming out? :grimacing: 

For the release, I am guessing successful RTD build is more important. :woman_shrugging: 

Also see #7943

### PR DESCRIPTION
**Pull request by pllim**
Fix #8051 (hopefully)

Follow up of #8041

Note to self: Check the numpy-dev build carefully!

### CODE REVIEW

File: pip-requirements [hunk 1]
```diff
@@ -1,2 +1,2 @@
-numpy>=1.10.0,<1.15.3
+numpy>=1.10.0,!=1.15.3
```
Comment (pllim) [inline pip-requirements]: @saimn , is it just `!` or `!=`?

**Comment by pllim** [comment]
At least `setup.py` fix works (sorry, should have just done it this way to begin with); all tests passed, except for that one CircleCI failure that is unrelated. Someone need to double check the `pip-requirements` syntax.

File: pip-requirements (hunk 1 above)
Comment (bsipocz) [reply pip-requirements]: it should be `!=`

**Comment by bsipocz** [comment]
thank you @pllim!

**Pull request merged**